src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java

```diff
@@ -192,43 +192,24 @@ public void deleteDirectoryEntriesFromCollection(Firestore firestore, String col
         DocumentSnapshot apply(Transaction xn) throws InterruptedException;
     }
 
-    // Returns null if not found - upper layers do any throwing
-    private FireStoreDirectoryEntry runTransactionWithRetry(Firestore firestore,
-                                                            LookupFunction lookupFunction,
-                                                            String transactionOp,
-                                                            String warnMessage) throws InterruptedException {
-        for (int i = 0; i < LOOKUP_RETRIES; i++) {
-            try {
-                ApiFuture<FireStoreDirectoryEntry> transaction =
-                    firestore.runTransaction(
-                        xn -> {
-                            DocumentSnapshot docSnap = lookupFunction.apply(xn);
-                            if (docSnap == null) {
-                                return null;
-                            }
-                            return docSnap.toObject(FireStoreDirectoryEntry.class);
-                        });
-
-                return fireStoreUtils.transactionGet(transactionOp, transaction);
-            } catch (Exception ex) {
-                if (FireStoreUtils.shouldRetry(ex)) {
-                    logger.warn("Retry-able error in firestore future get - " +
-                        warnMessage + " message: " + ex.getMessage());
-                } else {
-                    throw new FileSystemExecutionException(transactionOp + " execution exception", ex);
-                }
-            }
-
-            TimeUnit.SECONDS.sleep(1);
-        }
-        throw new FileSystemExecutionException(transactionOp + " failed - no more retries");
-    }
+    private static final int SLEEP_BASE_MILLISECONDS = 1000;
 
     // Returns null if not found - upper layers do any throwing
     public FireStoreDirectoryEntry retrieveById(
         Firestore firestore, String collectionId, String fileId) throws InterruptedException {
 
-        return runTransactionWithRetry(firestore, (xn) -> lookupByFileId(firestore, collectionId, fileId, xn),
+        return fireStoreUtils.runTransactionWithRetry(firestore,
+            new FireStoreUtils.FirestoreFunction() {
+                @Override
+                public FireStoreDirectoryEntry apply(Transaction xn)
+                    throws InterruptedException {
+                    DocumentSnapshot docSnap = lookupByFileId(firestore, collectionId, fileId, xn);
+                    if (docSnap == null) {
+                        return null;
+                    }
+                    return docSnap.toObject(FireStoreDirectoryEntry.class);
```

Comment:
This can be condensed to 
```
return Optional.ofNullable(docSnap).map(d -> docSnap.toObject(FireStoreDirectoryEntry.class))
                        .orElse(null);
```